src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -898,8 +898,17 @@ MetadataDropCommands(void)
 	dropSnapshotCommandList = list_concat(dropSnapshotCommandList,
 										  detachPartitionCommandList);
 
+	/*
+	 * We are most likely re-creating the metadata, so not lose
```

Comment:
As we use it only for recreating (or reactivating in near future) we can remove the 'most likely' part.